include/filters/realtime_circular_buffer.hpp

```diff
@@ -80,6 +80,16 @@ class RealtimeCircularBuffer
 
   void set_capacity(unsigned int order, const T & value);
 
+  auto begin()
```

Comment:
What about specifying the full iterator type instead of using `auto`? It would be both more friendly to older compilers (not sure that is an issue now) and to readers of the API (that is definitely an issue).

Also, please add a documentation comment extracting the interesting bits from the docs of Boost: that the iterator is not invalidated if the original element is thrown away by another element.

Reply:
Perfect, will implement ASAP. Thank you :)

Reply:
I did two things:
- switching to cbegin/cend (because I can't find any application of being able to modify elements in place w.r.t. just reading them).
- made explicit the return type